xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-api/src/main/java/org/xwiki/search/solr/internal/IndexingUserConfigurationInitializer.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.search.solr.internal;
+
+import javax.annotation.Priority;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.AbstractMandatoryDocumentInitializer;
+import com.xpn.xwiki.doc.MandatoryDocumentInitializer;
+import com.xpn.xwiki.doc.XWikiDocument;
+
+/**
+ * Initializes the configuration page for the indexing user.
+ *
+ * @version $Id$
+ * @since 14.8M1
+ */
+@Component
+@Named("XWiki.SolrSearchAdminIndexingUser")
+@Singleton
+// initalize after the corresponding class
+@Priority(MandatoryDocumentInitializer.DEFAULT_PRIORITY + 100)
+public class IndexingUserConfigurationInitializer extends AbstractMandatoryDocumentInitializer
```

Comment:
Since this document is currently only used on main wiki, the initializer (and the other one I guess) should probably reflect it. You can indicate this in #isMainWikiOnly().

Reply:
Yes, I could do it, but the UI Section is also show in the subwikis (currently with a "there is nothing to do here, sorry", message) as I could not find a way to only display it in the main wiki, so I thought to have the object there, too, just in case I figure out how to use this in subwikis. (Come to think about it, maybe I should move that UIX to an initializer, too?).
Well, I add the `isMainWikiOnly` in the two initializers here. This can be removed any time the indexing user can be made to work in subwikis.

Reply:
>  This can be removed any time the indexing user can be made to work in subwikis.

+1

Reply:
Ok, done.